src/blocks/libvirt.rs

```diff
@@ -73,51 +73,95 @@ impl ConfigBlock for Libvirt {
 
 impl Block for Libvirt {
     fn update(&mut self) -> Result<Option<Update>> {
-        if !self
+        // Check if the connection is still active and re-instatiate it if it's not
+        match self
             .qemu_conn
             .is_alive()
-            .expect("unrecoverable error with qemu object")
+            .block_error("virt", "error when getting connection status to libvirt")
         {
-            self.qemu_conn = Connect::open_read_only(
-                &self
-                    .qemu_conn
-                    .get_uri()
-                    .expect("could not get URI from the currently connected QEMU object"),
-            )
-            .expect("could not re-connect to libvirtd")
+            Ok(true) => {}
+            Ok(false) => {
+                self.qemu_conn = match Connect::open_read_only(
+                    &self
+                        .qemu_conn
+                        .get_uri()
+                        .block_error("virt", "error in retrieving URI information from libvirt")
+                        .unwrap(), /* Only happens if memory corruption; then we should panic anyway */
```

Comment:
Hmm, lets propagate this just in case in the future `.get_uri()` will return error for some other, less critical reasons.